src/core/optimization_container.jl

```diff
@@ -404,55 +380,49 @@ function get_constraint_names(optimization_container::OptimizationContainer)
     return collect(keys(optimization_container.constraints))
 end
 
-function assign_constraint!(
+# This is a temporary method while refactoring variable container
+function add_cons_container!(
     optimization_container::OptimizationContainer,
-    constraint_type::AbstractString,
-    ::Type{T},
-    value,
-) where {T <: PSY.Component}
-    assign_constraint!(
+    cons_type::ConstraintType,
+    var_key::VariableKey,
+    axs...;
+    sparse = false,
+)
+    return add_cons_container!(
         optimization_container,
-        make_constraint_name(constraint_type, T),
-        value,
+        cons_type,
+        var_key.entry_type(),
+        var_key.component_type,
```

Comment:
There is an odd asymmetry the type vs instance. I guess that's just the way it is. Could you pass only `var_key`?

Reply:
I asked @kdheepak about this and he pointed out that all the VariableTypes are generally used as instances in the code. I don't have a specific preference.

Reply:
Yes, I will make this change when refactoring the constraints container.